Doc/c-api/long.rst

```diff
@@ -529,10 +529,190 @@ distinguished from a number.  Use :c:func:`PyErr_Occurred` to disambiguate.
+
+   See also :attr:`sys.int_info` which exposes similar information to Python.
+
+   .. c:member:: uint8_t bits_per_digit
+
+      Bits per digit.
+
+   .. c:member:: uint8_t digit_size
+
+      Digit size in bytes.
+
+   .. c:member:: int8_t word_endian
+
+      Word endian:
+
+      - ``1`` for most significant byte first (big endian)
+      - ``-1`` for least significant first (little endian)
+
+   .. c:member:: int8_t array_endian
+
+      Array endian:
+
+      - ``1`` for most significant byte first (big endian)
+      - ``-1`` for least significant first (little endian)
+
+
+.. c:function:: PyObject* PyLong_Import(int negative, size_t ndigits, Py_digit *digits)
+
+   Create a Python :class:`int` object from an array of digits.
+
+   On success, return a Python :class:`int` object.
+   On error, set an exception and return ``NULL``.
+
+   *negative* is ``1`` if the number is negative, or ``0`` otherwise.
+
+   *ndigits* is the number of digits in the *digits* array.
+
+   *digits* is an array of unsigned digits. Digits must be in the range
+   [``0``; ``PyLong_BASE - 1``].
+
+   See :c:struct:`PyLong_LAYOUT` for the layout of an array of digits.
+
+   See also the :ref:`PyLongWriter API <pylongwriter>`.
+
+
+.. c:struct:: PyLong_DigitArray
+
+   A Python :class:`int` object exported as an array of digits.
+
+   See :c:struct:`PyLong_LAYOUT` for the :c:member:`digits` layout.
+
+   .. c:member:: PyLongObject *obj
+
+      Strong reference to the Python :class:`int` object.
+
+   .. c:member:: int negative
+
+      1 if the number is negative, 0 otherwise.
+
+   .. c:member:: size_t ndigits
+
+      Number of digits in :c:member:`digits` array.
+
+   .. c:member:: Py_digit *digits
+
+      Array of unsigned digits.
+
+
+.. c:function:: int PyLong_Export(PyObject *obj, PyLong_DigitArray *array)
+
+   Export a Python :class:`int` object as an array of digits.
+
+   On success, set *\*array* and return 0.
+   On error, set an exception and return -1.
+
+   This function always succeeds if *obj* is a Python :class:`int` object or a
+   subclass.
+
+   :c:func:`PyLong_ReleaseExport` must be called once done with using
+   *export*.
+
+
+.. c:function:: void PyLong_ReleaseExport(PyLong_DigitArray *array)
+
+   Release the export *array* created by :c:func:`PyLong_Export`.
+
+
+.. _pylongwriter:
+
+PyLongWriter API
+^^^^^^^^^^^^^^^^
+
+The :c:func:`PyLong_Import` function can be used to create a Python
+:class:`int` from an array of digits. When converting an integer from a format
+to a Python :class:`int`, creating a temporary array of digits can be
+inefficient. In this case, the :c:type:`PyLongWriter` API can be used for even
+more efficient import.
+
+.. versionadded:: 3.14
+
+.. c:struct:: PyLongWriter
+
+   A Python :class:`int` writer instance.
+
+   The instance must be destroyed by :c:func:`PyLongWriter_Finish`.
+
+
+.. c:function:: PyLongWriter* PyLongWriter_Create(int negative, size_t ndigits, Py_digit **digits)
```

Comment:
This function has two return values (writer and digits).  Is this a good price to hide PyLongWriter structure details?  May we want for PyLongWriter something different than PyLong_DigitArray?

Reply:
There is an optimization: there is no real PyLongWriter structure in fact, just a cast between pointer types. It's not uncommon for functions to return more than one value.

Structures can also make the maintenance and evolution of this API more complicated. Programming languages which cannot use C header files have to reimplement the structure, it's more work for them.